torch/_subclasses/fake_tensor.py

```diff
@@ -1934,6 +1950,24 @@ def go(t: object, real_t: Tensor) -> None:
                 else:
                     tree_map_(go, fake_out, real_out)
 
+                if isinstance(fake_out, torch.Tensor):
+                    assert fake_out.dtype == real_out.dtype, (
```

Comment:
There are a lot more metadata properties than this?

Reply:
Just updated, to follow what CrossRefFakeMode does